scripts/xtensa-build-zephyr.py

```diff
@@ -356,38 +294,39 @@ def create_zephyr_sof_symlink():
 			"security-policy-settings/create-symbolic-links")
 		raise
 
-def west_init_update():
-	"""[summary] Downloads zephyrproject inside sof/ and create a ../../.. symbolic
-	link back to sof/"""
+def west_init():
+	"""[summary] Checks whether west had been initialized to SOF project.
+	If not, performs initialization.
+	"""
 	global west_top, SOF_TOP
-	zephyr_dir = pathlib.Path(west_top, "zephyr")
-
-	z_remote = "origin"
-	z_ref = args.zephyr_ref
-
-	# Example of how to override SOF CI and point it at any Zephyr
-	# commit from anywhere and to run all tests on it. Simply
-	# uncomment and edit these lines and submit as an SOF Pull
-	# Request. Note: unlike many git servers, github allows direct
-	# fetching of (full 40-digits) SHAs; even SHAs not in origin but
-	# in forks! See the end of "git help fetch-pack".
-	#
-	# z_remote = "https://somewhere.else"
-	# z_ref = "pull/38374/head"
-	# z_ref = "cb9a279050c4e8ad4663ee78688d8e7de1cac953"
-
-	# SECURITY WARNING for reviewers: never allow unknown code from
-	# unknown submitters on any CI system.
-
-	execute_command(["git", "clone", "--depth", "5", args.url, str(zephyr_dir)], timeout=1200)
-	execute_command(["git", "fetch", "--depth", "5", z_remote, z_ref], timeout=300, cwd=zephyr_dir)
-	execute_command(["git", "checkout", "FETCH_HEAD"], cwd=zephyr_dir)
-	execute_command(["git", "-C", str(zephyr_dir), "--no-pager", "log", "--oneline", "--graph",
-		"--decorate", "--max-count=20"])
-	execute_command(["west", "init", "-l", str(zephyr_dir)])
-	create_zephyr_sof_symlink()
-	# Do NOT "west update sof"!!
-	execute_command(["west", "update", "zephyr", "hal_xtensa"], timeout=300, cwd=west_top)
+	result_rootdir = execute_command(["west", "topdir"], capture_output=True, cwd=west_top,
+		timeout=10, check=False)
+	if result_rootdir.returncode == 0:
+		result_manifest = execute_command(["west", "config", "manifest.path"], capture_output=True,
+			cwd=west_top, timeout=10, check=True)
+		west_root_dir = pathlib.Path(result_rootdir.stdout.decode().strip('\n'))
+		west_manifest_dir = pathlib.Path(west_root_dir, result_manifest.stdout.decode().strip('\n')).resolve()
+
+		if str(west_manifest_dir) != str(SOF_TOP.resolve()):
+			west_workspace_dir = pathlib.Path(west_root_dir, ".west")
+			message = "West is initialized to manifest other than SOFs!\n"
+			message += f"Initialized to manifest in: {west_manifest_dir} directory." + "\n"
+			message += f"Try removing {west_workspace_dir.resolve()} directory and rerun this script."
+			raise RuntimeError(message)
+		return
+	execute_command(["west", "init", "-l", str(SOF_TOP)], check=True)
+
+def west_update():
+	"""[summary] Downloads zephyr to west root directory and create symbolic
+	link from modules/audio/sof back to sof/"""
+	global west_top
+	execute_command(["west", "update"], check=True, timeout=3000, cwd=west_top)
+
+def west_update_submodules():
```

Comment:
```suggestion
def west_update_sof_rimage():
```
They're not "submodules" anymore here.

Reply:
I'm not convinced of the value of updating rimage _only_. Why not update hal_xtensa and all other SOF dependencies at the same time?